### DISCUSSION TITLE
How to create a 'locked' text field

### DISCUSSION THREAD

**Discussion by luiz00martins** [Pyroxenium/Basalt#31] [Q&A] [answered]
I want to display some text, in a way that the user cannot interact or modify. For example, the code below, this text field is modifiable:

```lua
main:addTextfield('textfield')
	:setPosition(1, 1)
	:setSize(20, 1)
	:addLine('Value')
	:addLine('Another')
```

https://user-images.githubusercontent.com/43142209/196071242-8fee485c-02e8-442b-b0ab-2280545e8fde.mp4

Is there a way to make it unmodifiable by the user? (i.e. display only)

**Comment by NoryiE** [comment, marked as answer]
Hi, if you don't want the user to modify your text you can use :disable():
https://basalt.madefor.cc/#/objects/Object/disable

If your goal is to disable writing only you could use something like this:
:onChar(function() return false end)
